Sources/FluidAudio/DownloadUtils.swift

```diff
@@ -1,289 +1,202 @@
         }
 
-        // Download subdirectory files required by CoreML
-        let subDirs = ["analytics", "model", "neural_network_optionals"]
-        for subDir in subDirs {
-            let subDirPath = folderPath.appendingPathComponent(subDir)
-            try FileManager.default.createDirectory(at: subDirPath, withIntermediateDirectories: true)
-
-            let subFileURL = "\(baseURL)/\(subDir)/coremldata.bin"
-            let subDestinationPath = subDirPath.appendingPathComponent("coremldata.bin")
+        // Configure CoreML
+        let config = MLModelConfiguration()
+        config.computeUnits = computeUnits
+
+        // Load each model
+        var models: [String: MLModel] = [:]
+        for name in modelNames {
+            let modelPath = repoPath.appendingPathComponent(name)
+            guard FileManager.default.fileExists(atPath: modelPath.path) else {
+                throw CocoaError(.fileNoSuchFile, userInfo: [
+                    NSFilePathErrorKey: modelPath.path,
+                    NSLocalizedDescriptionKey: "Model file not found: \(name)"
+                ])
+            }
+            
+            // Verify model size (CoreML models should be > 1KB at minimum)
+            if let attrs = try? FileManager.default.attributesOfItem(atPath: modelPath.path),
+               let fileSize = attrs[.size] as? Int64,
+               fileSize < 1024 {
+                logger.warning("⚠️ Model \(name) seems too small: \(fileSize) bytes")
+            }
 
             do {
-                try await downloadFile(from: subFileURL, to: subDestinationPath)
+                models[name] = try MLModel(contentsOf: modelPath, configuration: config)
+                logger.info("✅ Loaded model: \(name)")
             } catch {
-                // Some subdirectory files might be optional
+                logger.error("❌ Failed to load model \(name): \(error)")
+                throw error
             }
         }
+
+        return models
     }
 
-    /// Download a single file from URL
-    public static func downloadFile(from urlString: String, to destinationPath: URL) async throws {
-        guard let url = URL(string: urlString) else {
-            throw URLError(.badURL)
+    /// Download a HuggingFace repository using git clone
+    private static func downloadRepo(_ repo: Repo, to directory: URL) async throws {
+        logger.info("📥 Downloading \(repo.folderName) from HuggingFace...")
+        print("📥 Downloading \(repo.folderName)...")
+        
+        // Check if git is available
+        let gitPath = "/usr/bin/git"
+        guard FileManager.default.fileExists(atPath: gitPath) else {
+            logger.error("❌ Git not found at \(gitPath)")
+            throw URLError(.cannotFindHost, userInfo: [
+                NSLocalizedDescriptionKey: "Git is not installed. Please install Xcode Command Line Tools by running 'xcode-select --install' in Terminal."
+            ])
         }
+        
```

Comment:
Wait no I don think this will work. Majority of users will not have this installed